### ISSUE TITLE
Lotus can seal CC sectors but is not able to start AP for sectors with deals.

### ISSUE THREAD

**Issue by donkabat** [filecoin-project/lotus#9891]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [X] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [X] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
Daemon:  1.19.0+mainnet+git.64059ca87+api1.5.0
Local: lotus-miner version 1.19.0+mainnet+git.64059ca87
lotus-worker version 1.18.2+mainnet+git.9416ceb7d
boostd version 1.5.1-rc2+git.b5768c8
```

### Describe the Bug

I moved miner and boost and joined them on one server.
Miner's config dir copied manually between severs on version v1.18.0 and next upgraded to v1.19.0.
Boostd built and recovered from backup.

Lotus can seal CC sectors but is not able to start AP for sectors with deals.

```
cat lotus-miner/config.toml
PreferNewSectorsForDeals = true
```
I tried with indexer turned off and on.

### Logging Information

```text
cat lotus-miner.log
2022-12-15T14:17:30.146Z        INFO    sectors pipeline/input.go:286   Adding piece for deal 18872674 (publish msg: bafy2bzacebhik6omnx6crewpfv6grc6sdus6q6d2vin3fujbkppuunhn2bvoa)
2022-12-15T14:17:30.169Z        ERROR   sectors pipeline/input.go:575   we are trying to create a new sector with open sectors map[{1127678 25934}:0xc0c1ebd8c0 {1127678 27092}:0xc03d241aa0 {1127678 27351}:0xc04fe927e0 {1127678 31508}:0xc03e3b9980 {1127678 31962}:0xc03d7684b0 {1127678 32278}:0xc060d95140 {1127678 32291}:0xc06133c9f0]

cat boostd.log
2022-12-15T14:16:50.417Z        INFO    boost-storage-deal      logs/log.go:40  user initiated deal retry       {"id": "1eff5069-a798-4a75-9a8d-afd93ff59e82", "checkpoint": "PublishConfirmed"}
2022-12-15T14:16:50.418Z        INFO    boost-storage-deal      logs/log.go:40  deal execution initiated        {"id": "1eff5069-a798-4a75-9a8d-afd93ff59e82", "deal state": {"DealUuid":"1eff5069-a798-4a75-9a8d-afd93ff59e82","CreatedAt":"2022-12-15T13:32:15.029484699Z","ClientDealProposal":{"Proposal":{"PieceCID":{"/":"baga6ea4seaqa2nmvuefycv4f4bn5y42es633zekihqb7ywatxrglfoaq7zyd6pq"},"PieceSize":34359738368,"VerifiedDeal":true,"Client":"f1ieughpsgkdynelb2tipfpuhdvcvltfvsuv54q2q","Provider":"f01127678","Label":"bafybeifxitgxrp7w6nmvoefvbo7y7xkm77a55cmdadsxcbxoyrhasc7rgu","StartEpoch":2450000,"EndEpoch":3962000,"StoragePricePerEpoch":"0","ProviderCollateral":"10397687728628581","ClientCollateral":"0"},"ClientSignature":{"Type":0,"Data":null}},"IsOffline":false,"ClientPeerID":"12D3KooWSiA7jb5huuutQJ3DRBM7EUyNJznMAorbZNYNhGFwE3cj","DealDataRoot":{"/":"bafybeifxitgxrp7w6nmvoefvbo7y7xkm77a55cmdadsxcbxoyrhasc7rgu"},"InboundFilePath":"/home/filecoin/.boost/incoming/1eff5069-a798-4a75-9a8d-afd93ff59e82.download","Transfer":{"Type":"http","ClientID":"","Params":null,"Size":25771993853},"ChainDealID":18872672,"PublishCID":{"/":"bafy2bzacebhik6omnx6crewpfv6grc6sdus6q6d2vin3fujbkppuunhn2bvoa"},"SectorID":0,"Offset":0,"Length":0,"Checkpoint":3,"CheckpointAt":"2022-12-15T14:04:30.631133289Z","Err":"failed to add piece: packing piece baga6ea4seaqa2nmvuefycv4f4bn5y42es633zekihqb7ywatxrglfoaq7zyd6pq: AddPiece failed: RPCConnectionError","Retry":"auto","NBytesReceived":0,"FastRetrieval":false}}
2022-12-15T14:16:50.582Z        INFO    boost-storage-deal      logs/log.go:40  size of downloaded file {"id": "1eff5069-a798-4a75-9a8d-afd93ff59e82", "filepath": "/home/filecoin/.boost/incoming/1eff5069-a798-4a75-9a8d-afd93ff59e82.download", "size": 25771993853}
2022-12-15T14:16:50.587Z        INFO    boost-storage-deal      logs/log.go:40  deal execution in progress      {"id": "1eff5069-a798-4a75-9a8d-afd93ff59e82"}
2022-12-15T14:16:50.596Z        INFO    boost-storage-deal      logs/log.go:40  deal data transfer has already been completed   {"id": "1eff5069-a798-4a75-9a8d-afd93ff59e82"}
2022-12-15T14:16:50.605Z        INFO    boost-storage-deal      logs/log.go:40  deal data-transfer can no longer be cancelled   {"id": "1eff5069-a798-4a75-9a8d-afd93ff59e82"}
2022-12-15T14:16:50.611Z        INFO    boost-storage-deal      logs/log.go:40  deal has already been published and confirmed-publish   {"id": "1eff5069-a798-4a75-9a8d-afd93ff59e82"}
2022-12-15T14:16:50.619Z        INFO    boost-storage-deal      logs/log.go:40  funds successfully untagged for deal after publish      {"id": "1eff5069-a798-4a75-9a8d-afd93ff59e82"}
2022-12-15T14:16:50.641Z        INFO    boost-storage-deal      logs/log.go:40  add piece called        {"id": "1eff5069-a798-4a75-9a8d-afd93ff59e82"}
2022-12-15T14:17:22.002Z        INFO    boost-storage-deal      logs/log.go:40  user initiated deal retry       {"id": "b9ac8e8a-5608-439d-8311-9588867511b1", "checkpoint": "PublishConfirmed"}
2022-12-15T14:17:22.002Z        INFO    boost-storage-deal      logs/log.go:40  deal execution initiated        {"id": "b9ac8e8a-5608-439d-8311-9588867511b1", "deal state": {"DealUuid":"b9ac8e8a-5608-439d-8311-9588867511b1","CreatedAt":"2022-12-15T13:45:51.383843169Z","ClientDealProposal":{"Proposal":{"PieceCID":{"/":"baga6ea4seaqa2tcwoz3ttgedrbnrgvkuekzf4u4paxf2h4lc7uqkbmxenpwdipy"},"PieceSize":34359738368,"VerifiedDeal":true,"Client":"f1ieughpsgkdynelb2tipfpuhdvcvltfvsuv54q2q","Provider":"f01127678","Label":"bafybeiehkf4zwhvdh6m2xrbwd6mw6ix6vhusu5zbodpme4sjliw7mwbwxy","StartEpoch":2450000,"EndEpoch":3962000,"StoragePricePerEpoch":"0","ProviderCollateral":"10397647287464617","ClientCollateral":"0"},"ClientSignature":{"Type":0,"Data":null}},"IsOffline":false,"ClientPeerID":"12D3KooWSiA7jb5huuutQJ3DRBM7EUyNJznMAorbZNYNhGFwE3cj","DealDataRoot":{"/":"bafybeiehkf4zwhvdh6m2xrbwd6mw6ix6vhusu5zbodpme4sjliw7mwbwxy"},"InboundFilePath":"/home/filecoin/.boost/incoming/b9ac8e8a-5608-439d-8311-9588867511b1.download","Transfer":{"Type":"http","ClientID":"","Params":null,"Size":25771993853},"ChainDealID":18872673,"PublishCID":{"/":"bafy2bzacebhik6omnx6crewpfv6grc6sdus6q6d2vin3fujbkppuunhn2bvoa"},"SectorID":0,"Offset":0,"Length":0,"Checkpoint":3,"CheckpointAt":"2022-12-15T14:04:30.646030817Z","Err":"failed to add piece: packing piece baga6ea4seaqa2tcwoz3ttgedrbnrgvkuekzf4u4paxf2h4lc7uqkbmxenpwdipy: AddPiece failed: RPCConnectionError","Retry":"auto","NBytesReceived":0,"FastRetrieval":false}}
2022-12-15T14:17:22.101Z        INFO    boost-storage-deal      logs/log.go:40  size of downloaded file {"id": "b9ac8e8a-5608-439d-8311-9588867511b1", "filepath": "/home/filecoin/.boost/incoming/b9ac8e8a-5608-439d-8311-9588867511b1.download", "size": 25771993853}
2022-12-15T14:17:22.108Z        INFO    boost-storage-deal      logs/log.go:40  deal execution in progress      {"id": "b9ac8e8a-5608-439d-8311-9588867511b1"}
2022-12-15T14:17:22.115Z        INFO    boost-storage-deal      logs/log.go:40  deal data transfer has already been completed   {"id": "b9ac8e8a-5608-439d-8311-9588867511b1"}
2022-12-15T14:17:22.120Z        INFO    boost-storage-deal      logs/log.go:40  deal data-transfer can no longer be cancelled   {"id": "b9ac8e8a-5608-439d-8311-9588867511b1"}
2022-12-15T14:17:22.125Z        INFO    boost-storage-deal      logs/log.go:40  deal has already been published and confirmed-publish   {"id": "b9ac8e8a-5608-439d-8311-9588867511b1"}
2022-12-15T14:17:22.134Z        INFO    boost-storage-deal      logs/log.go:40  funds successfully untagged for deal after publish      {"id": "b9ac8e8a-5608-439d-8311-9588867511b1"}
2022-12-15T14:17:22.146Z        INFO    boost-storage-deal      logs/log.go:40  add piece called        {"id": "b9ac8e8a-5608-439d-8311-9588867511b1"}
2022-12-15T14:17:29.897Z        INFO    boost-storage-deal      logs/log.go:40  user initiated deal retry       {"id": "81d1fc90-6b4e-4301-94c9-d437ff751477", "checkpoint": "PublishConfirmed"}
2022-12-15T14:17:29.897Z        INFO    boost-storage-deal      logs/log.go:40  deal execution initiated        {"id": "81d1fc90-6b4e-4301-94c9-d437ff751477", "deal state": {"DealUuid":"81d1fc90-6b4e-4301-94c9-d437ff751477","CreatedAt":"2022-12-15T13:45:57.026344545Z","ClientDealProposal":{"Proposal":{"PieceCID":{"/":"baga6ea4seaqa2txlhofaqtdtpcagh6jyfed6p6g2bhhywjkmfyvewp522o6ecii"},"PieceSize":34359738368,"VerifiedDeal":true,"Client":"f1ieughpsgkdynelb2tipfpuhdvcvltfvsuv54q2q","Provider":"f01127678","Label":"bafybeiht2jgshq2nj6rrlcgb2jdwuc3rozn2rgdszgvizgdldr5cesriau","StartEpoch":2450000,"EndEpoch":3962000,"StoragePricePerEpoch":"0","ProviderCollateral":"10397647287464617","ClientCollateral":"0"},"ClientSignature":{"Type":0,"Data":null}},"IsOffline":false,"ClientPeerID":"12D3KooWSiA7jb5huuutQJ3DRBM7EUyNJznMAorbZNYNhGFwE3cj","DealDataRoot":{"/":"bafybeiht2jgshq2nj6rrlcgb2jdwuc3rozn2rgdszgvizgdldr5cesriau"},"InboundFilePath":"/home/filecoin/.boost/incoming/81d1fc90-6b4e-4301-94c9-d437ff751477.download","Transfer":{"Type":"http","ClientID":"","Params":null,"Size":25771993853},"ChainDealID":18872674,"PublishCID":{"/":"bafy2bzacebhik6omnx6crewpfv6grc6sdus6q6d2vin3fujbkppuunhn2bvoa"},"SectorID":0,"Offset":0,"Length":0,"Checkpoint":3,"CheckpointAt":"2022-12-15T14:04:30.660898449Z","Err":"failed to add piece: packing piece baga6ea4seaqa2txlhofaqtdtpcagh6jyfed6p6g2bhhywjkmfyvewp522o6ecii: AddPiece failed: RPCConnectionError","Retry":"auto","NBytesReceived":0,"FastRetrieval":false}}
2022-12-15T14:17:30.086Z        INFO    boost-storage-deal      logs/log.go:40  size of downloaded file {"id": "81d1fc90-6b4e-4301-94c9-d437ff751477", "filepath": "/home/filecoin/.boost/incoming/81d1fc90-6b4e-4301-94c9-d437ff751477.download", "size": 25771993853}
2022-12-15T14:17:30.096Z        INFO    boost-storage-deal      logs/log.go:40  deal execution in progress      {"id": "81d1fc90-6b4e-4301-94c9-d437ff751477"}
2022-12-15T14:17:30.105Z        INFO    boost-storage-deal      logs/log.go:40  deal data transfer has already been completed   {"id": "81d1fc90-6b4e-4301-94c9-d437ff751477"}
2022-12-15T14:17:30.114Z        INFO    boost-storage-deal      logs/log.go:40  deal data-transfer can no longer be cancelled   {"id": "81d1fc90-6b4e-4301-94c9-d437ff751477"}
2022-12-15T14:17:30.122Z        INFO    boost-storage-deal      logs/log.go:40  deal has already been published and confirmed-publish   {"id": "81d1fc90-6b4e-4301-94c9-d437ff751477"}
2022-12-15T14:17:30.129Z        INFO    boost-storage-deal      logs/log.go:40  funds successfully untagged for deal after publish      {"id": "81d1fc90-6b4e-4301-94c9-d437ff751477"}
2022-12-15T14:17:30.139Z        INFO    boost-storage-deal      logs/log.go:40  add piece called        {"id": "81d1fc90-6b4e-4301-94c9-d437ff751477"}
```

**Comment by piknikSteven2021** [comment]
We have not upgraded to v1.19 but we are also seeing this issue. In the logs, we consistently a spam of ERROR messages in the logs on multiple miners that are attempting to AddPiece some deals:
`/home/jenkins/build/workspace/lotus_v1.18.0-pk2/storage/pipeline/input.go:261
  - storage call error 0: pr read error: unexpected EOF [name: gpub-337167]: pr read error: unexpected EOF
2022-12-16T00:49:48.891Z        WARN    sectors pipeline/fsm.go:792     sector 15274 got error event sealing.SectorAddPieceFailed: writing piece: storage call error 0: pr read error: unexpected EOF [nam>2022-12-16T00:49:49.722Z        WARN    rpc     go-jsonrpc@v0.1.8/handler.go:329        error in RPC call to 'Filecoin.SectorAddPieceToAny': writing piece:
    github.com/filecoin-project/lotus/storage/pipeline.(*Sealing).handleAddPiece
        /home/jenkins/build/workspace/lotus_v1.18.0-pk2/storage/pipeline/input.go:261
  - storage call error 0: pr read error: unexpected EOF [name: gpub-337167]: pr read error: unexpected EOF
2022-12-16T00:49:49.722Z        WARN    sectors pipeline/fsm.go:792     sector 15470 got error event sealing.SectorAddPieceFailed: writing piece: storage call error 0: pr read error: unexpected EOF [nam>2022-12-16T00:49:52.284Z        ERROR   rpcenc  rpcenc/reader.go:408    context error in reader stream handler (2): context canceled
2022-12-16T00:49:52.284Z        ERROR   rpcenc  rpcenc/reader.go:408    context error in reader stream handler (2): context canceled`

After the context errors, it seems AddPiece completely fails to be scheduled anywhere.

**Comment by piknikSteven2021** [comment]
This build of v1.18 has the Snap Deal Aborted Sector fix and AddPiece fix patched into it.

**Comment by piknikSteven2021** [comment]
We'll be upgrading to v1.19 to test as well.

**Comment by TippyFlitsUK** [comment]
Thanks Steven, keep us posted!!

**Comment by donkabat** [comment]
I removed lotus-miner and restored them from backup but it didnt fix the issue.
Any new ideas what could I do?

**Comment by jcace** [comment]
Seeing this on my SP too, a bunch of sectors waiting in `Adding to Sector`

`lotus version 1.19.0`, `boostd version 1.5.0`

**Comment by piknikSteven2021** [comment]
We're still seeing AP not keeping up with the deals in StorageDealsStaged. Our StorageDealStaged has grown from 50 to 90 in an hour, and only 2 AddPiece at a time are actually running. We've recently extended 2000 sectors, so that shouldn't be an issue.

```
lotus@fc-lotus-miner:~$ lotus-miner sealing jobs
ID        Sector  Worker    Hostname     Task  State     Time
8c4f5db3  11620   86dbbd82  gpua-3371ae  PR2   running   8m41.7s
30c3db42  11625   9ec0f90a  gpua-33327f  PR2   running   8m19.2s
30d776dc  11615   3b108af3  gpua-33327f  PR2   running   8m16.3s
4e65fabd  11623   ab481300  gpua-3371d2  PR2   running   7m3.7s
02000c54  11612   cc5d151c  gpua-3371d2  PR2   running   6m25s
b0ca127a  11626   9f875254  gpua-3371cc  PR2   running   6m8.3s
8ab65a65  11640   a651cae4  gpua-33719b  RU    running   5m59.9s
95fb78ca  11649   8b4ed63a  gpua-33719b  PR2   running   4m2.6s
7213e482  11639   bd75775d  gpua-337248  PR2   running   3m48.1s
81763023  11643   2e83ad8d  gpua-3371a2  PR2   running   3m44s
d7b26b07  11634   a72b06eb  gpua-3371d2  RU    running   3m20.3s
5b19a5f0  11655   425698ad  gpua-337248  RU    running   3m15.5s
3c4aefe1  10175   89da68cd  gpua-3371cc  RU    running   2m58.8s
4c7a2200  11618   735b8aa2  gpua-33327f  RU    running   2m31.4s
bb7190fc  11632   f8f920b4  gpua-3371cc  GET   running   1m39.1s
1f3090e2  11648   2910a61b  gpua-337248  GET   running   1m38.5s
c9fee918  11621   9cc33125  pc1-c8b4fb   GET   running   1m36.7s
9028fb77  11631   58031d51  gpua-33327f  RU    running   1m36.1s
524d7fc6  10204   cd7e13aa  gpua-334383  RU    running   1m35.7s
c1b3c1c6  10167   9752ddfc  gpua-3371ae  RU    running   1m24.6s
0d2d5658  11617   93a93ca1  nas-7bea     GET   running   1m21.1s
aafb358e  11607   d8aa993c  gpua-3371ae  RU    running   1m11.4s
56538008  11635   70651aaf  gpua-3371a2  RU    running   1m7.6s
27d23c41  9959    086378a1  gpua-337248  PR2   running   53.4s
4751bfc1  11627   5f6b5626  gpua-334383  PR2   running   47.9s
8f7de39b  11636   a9fb3024  gpua-334383  RU    running   45.6s
12fedbeb  9059    18ba9d4d  gpua-3371a2  PR2   running   40.3s
eebfa1ad  11650   3818ab96  pc1-c881c5   AP    running   34.3s
b47d3fac  11628   a2b45cf5  pc1-c87be0   AP    running   34s
fd4a84db  8821    c5c5df90  gpua-3371ae  PR2   running   26s
a4820034  11598   e9904963  pc1-c8df89   GET   running   25.7s
90a880b4  11630   2bcc0e0c  gpua-33719b  PR2   running   25.3s
873b86c0  10213   ee5938ce  gpua-33719b  RU    running   24.5s
5cc47498  9957    80b37112  gpua-3371cc  PR2   running   14.4s
99bc84d6  10198   05cfcf56  gpua-3371d2  RU    running   11.5s
098573c0  11602   cd77c775  gpua-334383  PR2   running   6.9s
26ed633c  10209   aa9924e4  gpua-3371a2  RU    running   1.7s
00000000  10181   86dbbd82  gpua-3371ae  PR2   prepared  8m20.6s
00000000  9964    8b4ed63a  gpua-33719b  PR2   prepared  3m42.6s
00000000  11633   9ec0f90a  gpua-33327f  PR2   prepared  3m35.6s
00000000  8832    3b108af3  gpua-33327f  PR2   prepared  3m0.9s
00000000  8824    ab481300  gpua-3371d2  PR2   prepared  2m7.9s
00000000  11629   a651cae4  gpua-33719b  RU    prepared  1m49.4s
00000000  10430   c5c5df90  gpua-3371ae  PR2   prepared  3s
```
---
```
Storage Deals: 6468, 153.8 TiB
      Active:     6359  151.6 TiB (Verified: 6359 151.6 TiB)
      Publishing: 7     176 GiB   (Verified: 7    176 GiB)
      Publish:    7     160 GiB   (Verified: 7    160 GiB)
      Staged:     95    1.846 TiB (Verified: 95   1.846 TiB)
```
We have 26 workers configured to do AP, so there isn't really a reason for them not to. The AP workers don't show any issues in their logs. We'll leave it going over night, but we expect there to be either:
A) deals piling up in StorageDealStaged
B) APs stuck in Prepared


```
~$ lotus-miner version
Daemon:  1.19.0+mainnet+git.64059ca87+api1.5.0
Local: lotus-miner version 1.19.0+mainnet+git.64059ca87
```


```
root@pc1-c8e026:/home/lotus# lotus-worker --version
lotus-worker version 1.19.0+mainnet+git.64059ca87
root@pc1-c8e026:/home/lotus# LOTUS_WORKER_PATH=/home/lotus/ap-0 lotus-worker info
Worker version:  1.6.0
CLI version: lotus-worker version 1.19.0+mainnet+git.64059ca87

Session: b19cf3c8-db14-488d-83cf-52c672693fb4
Enabled: true
Hostname: pc1-c8e026
CPUs: 49; GPUs: []
RAM: 16.76 GiB/995.5 GiB; Swap: 0 B/0 B
Task types: FIN GET FRU C1 PR1 AP DC 

3af24e27-f492-5546-bf9e-86761fdd93a8:
        Weight: 1; Use: Seal 
        Local: /mnt/pc1-c8e026/scratch/storage
```

**Comment by piknikSteven2021** [comment]
To add onto this, I've looked at all the deals that have been stuck in Staged - the deals stuck range from now to the last 48 hours. Currently the miner is at 289 deals stuck in staged with 1-2 AddPiece happening at a time. The deals being selected are not following a queue
```
Storage Deals: 6660, 158.3 TiB
      Active:     6359  151.6 TiB (Verified: 6359 151.6 TiB)
      Publishing: 6     160 GiB   (Verified: 6    160 GiB)
      Publish:    6     144 GiB   (Verified: 6    144 GiB)
      Staged:     289   6.456 TiB (Verified: 289  6.456 TiB)
```
Additionally, looking at the deal status - the deals are immediately going into the StorageDealError state with the message handing off deal to node: packing piece at path fstmp1547957635: AddPiece failed: RPCConnectionError . This is only in regard to the ones that are making it through to AddPiece. We have many many deals stuck in Staged. Despite the error, it is evident that the deal was able to be sealed:
```
~$ lotus-miner sectors status 11557
SectorID:       11557
Status:         UpdateActivating
```
https://filfox.info/en/message/bafy2bzacebsuubewsgofmavctiq56yi3tyqrdotpn335eyzwjpnsmudogduzo
https://filfox.info/en/deal/19045411

**Comment by donkabat** [comment]
We spent with @TippyFlitsUK few hours trying to find a solution to the problem but we have to give up.
We initialized boostd without backup restoration and restored miner from backup instead of just copy dir of lotus-miner.

**Comment by donkabat** [comment]
I moved boost from server with CPU without Intel SHA Extensions to another one which support SHA to check if 
`RemoteCommp = false` will help... but no.

**Comment by jcace** [comment]
Still having this issue, unable to get the sectors past `AddPiece`. Boost has 85 sectors sitting in `Adding to Piece` right now, meanwhile my sealing pipeline is empty.

Lotus: v1.19.0
Boost: v.1.5.0
4 AP workers

No major changes to `config.toml` anywhere, though I have set `LOTUS_SEALING_MAKENEWSECTORFORDEALS="true"`. This is very strange as I was sealing fine for several days (weeks) before this without any problem, until it just stopped all of a sudden.

Upon restarting boost, I see these logs come though on the `miner`, as it called `AddPiece` again for everything:
```
{"level":"warn","ts":"2023-01-02T18:17:27.424-0800","logger":"rpc","caller":"go-jsonrpc@v0.1.8/handler.go:329","msg":"error in RPC call to 'Filecoin.SectorAddPieceToAny': context canceled"}
<…truncated messages for readability - 85 of these in total…>

{"level":"error","ts":"2023-01-02T18:17:27.688-0800","logger":"rpcenc","caller":"rpcenc/reader.go:408","msg":"context error in reader stream handler (2): context canceled"}
<…truncated messages for readability - 85 of these in total…>

{"level":"info","ts":"2023-01-02T18:17:35.951-0800","logger":"sectors","caller":"pipeline/input.go:292","msg":"Adding piece for deal 20778639 (publish msg: bafy2bzaceaep7a57l3dh4j74jcnjlwfhwpgdishpnardmyyyqz5r73pwhvjyw)"}
<…truncated messages for readability - 85 of these in total…>
```

**Comment by jcace** [comment]
I got AP's going! After a fresh restart of everything, I saw this error in the log:

```
{"level":"error","ts":"2023-01-02T21:02:04.311-0800","logger":"sectors","caller":"pipeline/input.go:580","msg":"we are trying to create a new sector with open sectors map[{1963614 3913}:0xc031039950 {1963614 3914}:0xc01ca046f0 {1963614 3915}:0xc001260720]"}
```

Which appears to be coming from [updateInput](https://github.dev/filecoin-project/lotus/blob/master/storage/pipeline/input.go#L439), which is called from addPendingPiece inside SectorAddPieceToAny ([here](https://github.dev/filecoin-project/lotus/blob/master/storage/pipeline/input.go#L353)). I noticed that this tries to assign the deal to "open sectors" (WaitDeals sectors?), so...

I tried removing the sector 3914, which was in `WaitDeals` state (refer to my screenshot in previous comment). 
`lotus-miner sectors remove --really-do-it 3914`

After that, the AP's started to happen again. Back to normal. 

Perhaps this is a different issue altogether, but it would seem that something can break this `updateInput` code if there are some sectors in WaitDeals.

TLDR: Had sectors in `WaitDeals`. Ran `lotus-miner sectors remove --really-do-it <sector>`. AP started again.

**Comment by jcace** [comment]
Just happened again. No AP's going through, stuck doing nothing. I had two sectors in the `WaitDeals` state, one with `15GiB` and one with `16GiB`. `remove`d the `16GiB` sector and AP started again. 

logs (sector `4700` was removed) and it kicked off again:
```
{"level":"error","ts":"2023-01-05T14:42:46.775-0800","logger":"sectors","caller":"pipeline/input.go:580","msg":"we are trying to create a new sector with open sectors map[{1963614 3914}:0xc01ca046f0 {1963614 4665}:0xc028c646f0 {1963614 4700}:0xc019b907e0]"}
{"level":"info","ts":"2023-01-05T14:42:46.776-0800","logger":"sectors","caller":"pipeline/input.go:813","msg":"new deal sector decision","sealing":38,"maxSeal":0,"maxUpgrade":0,"preferNew":false,"canCreate":true,"canUpgrade":true,"shouldUpgrade":true}
{"level":"info","ts":"2023-01-05T14:42:46.781-0800","logger":"stores","caller":"paths/remote.go:349","msg":"Delete http://10.32.32.5:12201/remote/unsealed/s-t01963614-4700"}
{"level":"warn","ts":"2023-01-05T14:42:46.797-0800","logger":"advmgr","caller":"sealer/manager.go:365","msg":"stub NewSector"}
{"level":"info","ts":"2023-01-05T14:42:46.797-0800","logger":"sectors","caller":"pipeline/input.go:839","msg":"Creating sector","number":"4701","type":"deal","proofType":8}
{"level":"error","ts":"2023-01-05T14:42:46.825-0800","logger":"sectors","caller":"pipeline/input.go:580","msg":"we are trying to create a new sector with open sectors map[{1963614 3914}:0xc01ca046f0 {1963614 4665}:0xc028c646f0 {1963614 4700}:0xc019b907e0 {1963614 4701}:0xc01bc1a330]"}
```

**Comment by jcace** [comment]
Narrowed down the issue. It arises when: 

- I have 2 sectors open in WaitDeals, waiting for more data 
- My miner config `MaxWaitDealsSectors = 2`
- New piece to be added is a full 32G, but Lotus won't create new sectors because of `MaxWaitDealsSectors`

Besides changing my `MaxWaitDealsSectors` to be bigger, I think it would make sense for Lotus to always create a new sector when the new sector will result in maximum QAP . Otherwise this situation arises where large pieces are not sealed until another smaller piece comes along to fill the space in the `WaitDeals` sector. 

### code links
- updateInput is unable to match the new sector to any existing ones since it's too big, so it ends up being left over in `toAssign` https://github.com/filecoin-project/lotus/blob/e68b8e96c7c28f4ff65f4b64254717549d158e87/storage/pipeline/input.go#L574
- this calls `tryGetDealSector`, which will not create a new sector because of the constraint on MaxWaitDealsSectors 
https://github.com/filecoin-project/lotus/blob/e68b8e96c7c28f4ff65f4b64254717549d158e87/storage/pipeline/input.go#L791

**Comment by peterliang270** [comment]
I also encountered the problem of boost issuing orders and miner card AP. Is there any good solution? The error message is as follows:
/provider.go:947 error：failed to addPiece for deal, will-retry  /mnt/md0/lotusworker/car/baga6ea4seaqezjlabwfrj3xvr3misoradvruwcyvt44fyqwqyfnbef54rv34ecy.car cc2e98d3-6c8a-4870-afa2-4deee2e4cc30 bafy2bzacecox2dllynr7ree7vyizrl2xapufkhvabjlq4zkenlf3c5phlvbo4 RPCConnectionError
